### PR TITLE
feat(cache): use new resource implementation

### PR DESCRIPTION
**Pull request by gcusnieux**
fix: gravitee-io/issues#599

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,11 +35,12 @@
 
     <properties>
         <gravitee-gateway-api.version>1.21.2</gravitee-gateway-api.version>
+        <gravitee-gateway.version>3.5.4</gravitee-gateway.version>
```
Comment (ytvnr) [inline pom.xml:38]: why not 3.7.0-SNAPSHOT ?

File: src/main/java/io/gravitee/policy/cache/CacheResponse.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 import io.gravitee.common.http.HttpHeaders;
 import io.gravitee.gateway.api.buffer.Buffer;
+import java.io.Serializable;
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/cache/CacheResponse.java:20]: you can delete this useless import

File: src/main/java/io/gravitee/policy/cache/mapper/CacheResponseMapper.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            addSerializer(Buffer.class, new BufferSerializerModule());
+        }
+    }
+
+    private class BufferDeserializerModule extends JsonDeserializer<Buffer> {
+
+        BufferFactory factory = new BufferFactoryImpl();
+
+        @Override
+        public Buffer deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/cache/mapper/CacheResponseMapper.java:58]:
```suggestion
        public Buffer deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
```

File: src/main/java/io/gravitee/policy/cache/resource/CacheElement.java [hunk 4]
```diff
@@ -26,13 +26,13 @@
 
     private final String key;
 
-    private final CacheResponse response;
+    private final Serializable object;
 
     private int timeToLive = 0;
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/cache/resource/CacheElement.java:31]: why not have a long here ? it could avoid a cast in CachePolicy.
wdyt ?

**Review by ytvnr — CHANGES_REQUESTED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**